test/framework/src/main/java/org/elasticsearch/transport/nio/SocketEventHandler.java

```diff
@@ -57,8 +58,8 @@ public void handleRegistration(NioSocketChannel channel) {
      * @param channel that was registered
      * @param exception that occurred
      */
-    public void registrationException(NioSocketChannel channel, Exception exception) {
-        logger.trace("failed to register channel", exception);
+    void registrationException(NioSocketChannel channel, Exception exception) {
+        logger.debug(new ParameterizedMessage("failed to register socket channel: {}", channel), exception);
```

Comment:
Since this is at the debug level, should you use a supplier here to avoid instantiating the parameterized message if not needed?